infoschema/tables.go

```diff
@@ -560,7 +560,7 @@ var partitionsCols = []columnInfo{
 	{name: "TABLESPACE_NAME", tp: mysql.TypeVarchar, size: 64},
 	{name: "TIDB_PARTITION_ID", tp: mysql.TypeLonglong, size: 21},
 	{name: "TIDB_PLACEMENT_POLICY_NAME", tp: mysql.TypeVarchar, size: 64},
-	{name: "TIDB_DIRECT_PLACEMENT", tp: mysql.TypeVarchar, size: types.UnspecifiedLength},
+	{name: "TIDB_DIRECT_PLACEMENT", tp: mysql.TypeVarchar, size: 512},
```

Comment:
Same here, I'm not sure if there actually is a maximum for Direct Placement, even though 512 would cover most cases, there might be more complex ones that would not be shown.

Reply:
ditto.

Reply:
Same here: 512.